pkg/sbom/telemetry/telemetry.go

```diff
@@ -82,3 +85,125 @@ var (
 		commonOpts,
 	)
 )
+
+// QueueMetricProvider is a workqueue.MetricsProvider that provides metrics for the SBOM queue
+type QueueMetricProvider struct{}
+
+// Ensure QueueMetricProvider implements the workqueue.MetricsProvider interface
+var _ workqueue.MetricsProvider = QueueMetricProvider{}
+
+type gaugeWrapper struct {
+	telemetry.Gauge
+}
+
+// Inc implements the workqueue.GaugeMetric interface
+func (g gaugeWrapper) Inc() {
+	g.Gauge.Inc()
+}
+
+// Dec implements the workqueue.GaugeMetric interface
+func (g gaugeWrapper) Dec() {
+	g.Gauge.Dec()
+}
+
+// Set implements the workqueue.GaugeMetric interface
+func (g gaugeWrapper) Set(v float64) {
+	g.Gauge.Set(v)
+}
+
+type counterWrapper struct {
+	telemetry.Counter
+}
+
+// Inc implements the workqueue.CounterMetric interface
+func (c counterWrapper) Inc() {
+	c.Counter.Inc()
+}
+
+type histgramWrapper struct {
+	telemetry.Histogram
+}
+
+// Observer implements the workqueue.HistogramMetric interface
+func (l histgramWrapper) Observe(value float64) {
+	l.Histogram.Observe(value)
+}
+
+// NewDepthMetric creates a new depth metric
+func (QueueMetricProvider) NewDepthMetric(string) workqueue.GaugeMetric {
+	return gaugeWrapper{telemetry.NewGaugeWithOpts(
+		subsystem,
+		"queue_depth",
+		[]string{},
+		"SBOM queue depth",
+		commonOpts,
+	)}
+}
+
+// NewAddsMetric creates a new adds metric
+func (QueueMetricProvider) NewAddsMetric(string) workqueue.CounterMetric {
+	return counterWrapper{telemetry.NewCounterWithOpts(
+		subsystem,
+		"queue_adds",
+		[]string{},
+		"SBOM queue adds",
+		commonOpts,
+	)}
+}
+
+// NewLatencyMetric creates a new latency metric
+func (QueueMetricProvider) NewLatencyMetric(string) workqueue.HistogramMetric {
+	return histgramWrapper{telemetry.NewHistogramWithOpts(
+		subsystem,
+		"queue_latency_seconds",
```

Comment:
I'm not sure we should add the unit in the metric name,

Reply:
Removed it